File: src/main/java/site/gachontable/gachontablebe/domain/admin/presentation/dto/response/WaitingInfosResponse.java
package site.gachontable.gachontablebe.domain.admin.presentation.dto.response;

import lombok.Builder;
import site.gachontable.gachontablebe.domain.shared.Table;
import site.gachontable.gachontablebe.domain.waiting.domain.Waiting;
import site.gachontable.gachontablebe.domain.waiting.type.Status;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record WaitingInfosResponse(Integer count, List<WaitingInfosResponse.WaitingInfo> waitingInfos) {

    @Builder
    public record WaitingInfo(String username, LocalDateTime time, Table tableType, String tel, UUID waitingId, Status waitingStatus) {
        public static WaitingInfo of(String userName, Waiting waiting) {
            return WaitingInfo.builder()
                    .username(userName)
                    .time(waiting.getCreatedAt())
                    .tableType(waiting.getTableType())
                    .tel(waiting.getTel())
                    .waitingId(waiting.getWaitingId())
                    .waitingStatus(waiting.getWaitingStatus())
                    .build();
        }
    }
}

File: src/main/java/site/gachontable/presentation/admin/dto/response/WaitingInfosResponse.kt
package site.gachontable.presentation.admin.dto.response

import site.gachontable.domain.waiting.domain.Waiting
import site.gachontable.domain.waiting.type.Status
import site.gachontable.presentation.shared.Table
import java.time.LocalDateTime
import java.util.*

data class WaitingInfosResponse(
    val count: Int,
    val waitingInfos: MutableList<WaitingInfo>,
) {
    data class WaitingInfo(
        val username: String,
        val time: LocalDateTime,
        val tableType: Table,
        val tel: String,
        val waitingId: UUID,
        val waitingStatus: Status,
    ) {
        companion object {
            fun of(username: String, waiting: Waiting): WaitingInfo {
                return WaitingInfo(
                    username = username,
                    time = waiting.createdAt,
                    tableType = waiting.tableType,
                    tel = waiting.tel,
                    waitingId = waiting.waitingId,
                    waitingStatus = waiting.waitingStatus
                )
            }
        }
    }
}
